ml/src/main/scala/frameless/ml/TypedTransformer.scala

```diff
@@ -0,0 +1,42 @@
+package frameless
+package ml
+
+import frameless.ops.SmartProject
+import org.apache.spark.ml.Transformer
+import shapeless.{Generic, HList}
+import shapeless.ops.hlist.{Prepend, Tupler}
+
+sealed trait TypedTransformer
+
+/**
+  * An AppendTransformer `transform` method takes as input a TypedDataset containing `Inputs` and
+  * return a TypedDataset with `Outputs` columns appended to the input TypedDataset.
+  */
+trait AppendTransformer[Inputs, Outputs, InnerTransformer <: Transformer] extends TypedTransformer {
```

Comment:
@atamborrino do we need to have our own Append operation here? Shapeless provides an operation that may do exactly this.

Reply:
`AppendTransformer` does a little more than just appending by first checking that `T` in `.transform[T]` can be projected to `Inputs`, and then indeed use shapeless's Prepend operation (see `i3: Prepend.Aux[TVals, OutputsVals, OutVals]`). The type-level append logic is basically the same than in 'TypedDataset.withColumn'.